haproxy/tests/conftest.py

```diff
@@ -59,45 +59,38 @@ def wait_for_haproxy():
 
 
 @pytest.fixture(scope="session")
-def spin_up_haproxy():
-    env = os.environ
-    env['HAPROXY_CONFIG_DIR'] = os.path.join(common.HERE, 'compose')
-    env['HAPROXY_CONFIG'] = os.path.join(common.HERE, 'compose', 'haproxy.cfg')
-    env['HAPROXY_CONFIG_OPEN'] = os.path.join(common.HERE, 'compose', 'haproxy-open.cfg')
-    env['HAPROXY_SOCKET_DIR'] = common.UNIXSOCKET_DIR
-    if Platform.is_linux() and not os.path.exists(common.UNIXSOCKET_DIR):
-        # make the temp directory on linux
-        os.makedirs(common.UNIXSOCKET_DIR)
-    args = [
-        "docker-compose",
-        "-f", os.path.join(common.HERE, 'compose', 'haproxy.yaml')
-    ]
-    subprocess.check_call(args + ["down"], env=env)
-    subprocess.check_call(args + ["up", "-d"], env=env)
-    wait_for_haproxy()
-    # subprocess.check_call(["ls", "-al", "/tmp/"], env=env)
-    # subprocess.check_call(["ls", "-al", "/tmp/haproxy"], env=env)
+def haproxy_container():
     try:
-        if Platform.is_linux():
+        env = os.environ
+        host_socket_dir = os.path.realpath(tempfile.mkdtemp())
+        host_socket_path = os.path.join(host_socket_dir, 'datadog-haproxy-stats.sock')
+
+        env['HAPROXY_CONFIG_DIR'] = os.path.join(common.HERE, 'compose')
+        env['HAPROXY_CONFIG'] = os.path.join(common.HERE, 'compose', 'haproxy.cfg')
+        env['HAPROXY_CONFIG_OPEN'] = os.path.join(common.HERE, 'compose', 'haproxy-open.cfg')
+        env['HAPROXY_SOCKET_DIR'] = host_socket_dir
+
+        args = [
+            "docker-compose",
+            "-f", os.path.join(common.HERE, 'compose', 'haproxy.yaml')
+        ]
+        subprocess.check_call(args + ["down"], env=env)
+        subprocess.check_call(args + ["up", "-d"], env=env)
+        wait_for_haproxy()
+        try:
             # on linux this needs access to the socket
             # it won't work without access
             chown_args = []
             user = getpass.getuser()
             if user != 'root':
                 chown_args += ['sudo']
-            chown_args += [
-                "chown", user, common.UNIXSOCKET_PATH
-            ]
+            chown_args += ["chown", user, host_socket_path]
```

Comment:
is `os.chmod()` not working ?

Reply:
nop and I tried.... a lot (see my commits)

Reply:
I believe it is because I need `sudo` (may be wrong)